src/Adapter/Alias/Validate/AliasValidator.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Alias\Validate;
+
+use Alias;
+use PrestaShop\PrestaShop\Adapter\AbstractObjectModelValidator;
+use PrestaShop\PrestaShop\Core\Domain\Alias\Exception\AliasConstraintException;
+use PrestaShop\PrestaShop\Core\Exception\CoreException;
+
+/**
+ * Validates alias field using legacy object model
+ */
+class AliasValidator extends AbstractObjectModelValidator
+{
+    /**
+     * This method is specific for alias creation only.
+     *
+     * @param Alias $alias
+     *
+     * @throws CoreException
+     */
+    public function validate(Alias $alias): void
+    {
+        $this->validateAliasProperty($alias, 'search', AliasConstraintException::INVALID_SEARCH);
+        $this->validateAliasProperty($alias, 'alias', AliasConstraintException::INVALID_ALIAS);
```

Comment:
I believe you need to check that the `alias` value doesn't exist already in the DB, we must ensure its unicity At least on creation of course, so maybe you need a dedicated validate method for creation

Reply:
Yes, you're right. Though though after small grooming we decided not to throw the error message if there is a duplicate `alias`, but instead skip it and create others (we could still log a warning to the Prestashop logger).

The skip already existing `alias` implementation is here: 
https://github.com/PrestaShop/PrestaShop/pull/32055/files#diff-95c43f0b3960c6c8b9565a24211b7fa2aaafbf18178ab8f53cb28844c721603fR84

The reasons to avoid throwing an error message:
Because merchant could add dozens of new aliases and then for example first 8 aliases will be created and others will be not created. Then when removing 9th alias and trying to resave, first 8 aliases will throw an error, etc.